### ISSUE TITLE
ES6 vs ES2015

### ISSUE THREAD

**Issue by anicolao** [getify/You-Dont-Know-JS#1883]
I already searched for this issue by reviewing all open issues, and searching closed issues. This is a duplicate of these "closed" issues:

https://github.com/getify/You-Dont-Know-JS/issues/1657
https://github.com/getify/You-Dont-Know-JS/issues/1594

The answer given is unsatisfying and the text as it stands is poorly written. The book itself is not at all clear about what the "correct" names are versus the "incorrect" names and the sanctimonious style in which it is at one juncture proclaimed that "JS6" is an incorrect name only to have the reader encounter "ES6" a few short sections later is off-putting.

Fix the text to be clear about what correct names are, or get off your high horse and let people use whatever names they like. The English language evolves due to misconceptions and incorrect usage all the time, and the distracting religion around naming detracts from the point of the chapter - or if it must be there, at least make it correct, and not just technically correct, but understandable to someone who doesn't know the history of exactly when it became incorrect to refer to the spec by its edition, which continues to be in the spec URLs to this day (as evidenced by ECMA-262, 15th edition at https://262.ecma-international.org/15.0/). 

If you really feel that there's a clear consensus that since 2016 nobody should refer to ECMAScript by which edition of the language specification they mean, then a reasonable alternate note would be:

NOTE:
--
It's become confusing use terms like "JS6" or "ES8" to refer to the language by which edition of the standard "6th edition" or "8th edition" is meant. Prior to 2016, this was common practice, but it stopped after ES6; so use the year to refer to the language specification for editions beyond ES6, by writing ES20xx, where ES2016 corresponds to the 7th edition and ES2024 corresponds to the 15th edition. Prior to ES2016, the correct names are ES6, ... (fill in the remainder for the reader).

**Comment by getify** [comment]
> get off your high horse

You're awfully upset about a brief side-note callout of the chapter. It's not like I spent 4 full paragraphs criticizing readers for using the wrong name of the language, as you've done here in critique of me. I don't know why that note offended you so strongly, but... it was really meant as a somewhat informal tip/advice (note the slightly relaxed language/tone), not a prolonged lecture.

If I was serious about correcting people and changing their behavior, I'd have dedicated specific paragraphs/sections in the book to the topic, not just a one-liner NOTE.

> at least make it correct

If you're insisting on the most explicit and strictly correct name, [the specification names itself](https://262.ecma-international.org/15.0/#sec-intro): "This Ecma Standard defines the ECMAScript 2024 Language."

So "ECMAScript 2024" with exactly that capitalization, spelling, and spacing, is the only strictly proper name for the language, as it relates to the code that people write and run in browsers or Node.js, etc.

That said, "ECMAScript" is commonly abbreviated as "ES", and the space is commonly elided, so "ES2024" would be how reasonable people refer to the correct/official name of the current language version. In fact, later in that same linked spec Intro section, you'll see it uses the specific abbreviated term "ES2016" (instead of "ECMAScript 2016"), so the spec is also clearly blessing/nodding-to this convention of abbreviation (and space elision) -- though there are no other similar terms present ("ES2015", "ES6", etc)... all other occurrences fully spell it out as "ECMAScript XXXX". Really, shame on them for being so inconsistent as to abbreviate in some places and not in others! :)

Also, readers of an intro book on JS aren't going to care about all those nuances anyway, so that'd just be distracting.

> The answer given is unsatisfying and the text as it stands is poorly written

I stand by [the explanation I gave here](https://github.com/getify/You-Dont-Know-JS/issues/1657#issuecomment-619445151), which basically in summary says:

> We're not changing historical naming, only fixing what's going forward.

### Further detail

There's no question ES3 and ES5 are what the language name was, officially and commonly, back in 1999-20014. And there's no question -- look up the old specifications from 2016 to 2024 -- that the *new* naming convention is "ECMAScript XXXX" and abbreviated "ESXXXX".

The *only* language version which is even remotely controversial is ES6 vs ES2015. It's historical fact that this change to "ECMAScript 2015" / "ES2015" didn't happen until only a few months before the official specification was finalized, after **YEARS** (and countless drafts) being named "ES6". And at that point, the majority of people who were anticipating this version landing -- it was in progress for 4+ years! -- knew it by "ES6", and wrote countless books and blogs and even ran conferences and printed tshirts (not kidding!), all with the name "ES6".

It would have been a huge disservice (of confusion) to the community to ***FORCE*** that change on folks so late. TC39 was wrong to shoehorn in that change to the spec last minute, but they knew nobody was going to seriously change all those prior ES6 usages. And I, being one of those published book authors with the name "ES6" already out there, decided to resist that last minute change. There was absolutely no such argument to be made supporting the folks who kept using "ES7", "ES8", etc, -- as they were doing *STILL* in 2019 when I wrote the book -- long after ES2016 had landed. None of those terms had been commonly and widely in use prior to or around the time of ES6 landing, so they came along *after* the naming change. All TC39 cared about, really, was they wanted to influence people to use the proper naming going forward, starting with ES2016, so I was backing them up on that -- though I still personally think the name change was pointless.

And BTW, "JS6" type naming was from Mozilla in like 1997 or whatever, so it was like 20 years out of date (and also confused with Microsoft's abomination of JScript 1 and 2, etc), and long since definitely should have been abandoned.

----

Anyway, while this trip down history lane has been amusing, it's completely unproductive.

> If you really feel that there's a clear consensus that since 2016

There absolutely is, and it's indisputable by historical fact.

> then a reasonable alternate note would be

We're arguing about a term that only applies to a language version from **9 years ago**. This is silly and trivial. I can't even remember the last time I needed to use the term "ES6", but it was probably *in this book*. We don't hardly ever refer to ES6 anymore, because it's so ubiquitous. The baseline "old language" version is probably ES2021 now, maybe ES2022. Nobody cares what was in ES6 or ES2016 anymore.

But the main reason now, aside from the points I've made thus far, that I am not going to make the *demanded* change, however well-meaning your (somewhat aggressive toned) feedback is:

***The book in question was published in print almost 5 years ago. I've literally sold more than 40K physical copies of it.***

I'm simply not making *any* changes to this book's manuscript text anymore, at all. I'm not fixing typos, and I'm not second guessing the wording of how I gave people an informal tip in a side note. And I'm not issuing errata updates to Amazon and other publishers to change their print runs. The book is frozen in time, with all my mistakes in clear black and white.

**Exception:** I would still make fixes on the repo if there were things like broken links (where URLs died or went away), to help people as they click on things while reading digitally here. But that's basically the only change I'd entertain this far down the line.

**Comment by anicolao** [comment]
I came back to this to delete my issue, but about 30m too late. Sorry to have put you to the work of writing such a long response.

My original bug report was written in a poorly thought through attempt to emulate your own aggressive writing style, as the prior bug reports were essentially ignored. I think you missed the main point of my bug report which is that the text as written seems to contradict itself and puts the reader off continuing to read. It's great that you've sold a lot of copies; that doesn't mean the text is perfect. If I was irritated, it was mainly by the fact that this had in fact already been reported twice and brushed off, even though it is one of the reader's first experiences with your text and you've got clear feedback that it's confusing. As the author, you ought to be more interested in why your text comes off as wrong rather than just defending it.

As for the state of the book, I had (mis-)understood from CONTRIBUTING.md that you consider the 2nd Ed a work in progress and are interested in contributions. Since you're not, I recommend you consider updating that document.

In any event, my apologies if I annoyed you, and I'm sorry I wasted my time and yours on your book.
